packages/formspree-react/src/context.tsx

```diff
@@ -1,98 +1,75 @@
+  children: ReactNode;
   project?: string;
-  children: React.ReactNode;
   stripePK?: string;
 };
 
-const FormspreeContext = React.createContext<FromspreeContextType | null>(null);
-
-FormspreeContext.displayName = 'Formspree';
-
-let stripePromise: Promise<Stripe | null>;
-
-const getStripe = (stripeKey: string) => {
-  if (!stripePromise) {
-    stripePromise = loadStripe(stripeKey);
-  }
-  return stripePromise;
-};
-
-const handleCreateClient = (promise?: Stripe, project?: string) => {
-  const config: Config = {};
-
-  if (promise) {
-    config.stripePromise = promise;
-  }
-
-  if (project) {
-    config.project = project;
-  }
-
-  return createClient(config);
-};
-
-export const FormspreeProvider = (props: FormspreeProviderProps) => {
-  const [stateStripePromise, setStateStripePromise] = useState<
-    Stripe | undefined
-  >(undefined);
-  const [client, setClient] = useState<Client>(
-    handleCreateClient(stateStripePromise, props.project)
+const FormspreeContext = React.createContext<FormspreeContextType | null>(null);
+
+/**
+ * FormspreeProvider creates Formspree Client based on the given props
+ * and makes the client available via context.
+ */
+export function FormspreeProvider(props: FormspreeProviderProps) {
+  const { children, project, stripePK } = props;
+  const [client, setClient] = useState(createClient({ project }));
+  const stripePromise = useMemo(
+    () => (stripePK ? loadStripe(stripePK) : null),
+    [stripePK]
   );
 
   useEffect(() => {
-    const getStripePromise = async (stripeKey: string) => {
-      const promiseStripe = await getStripe(stripeKey);
-      if (promiseStripe) {
-        setStateStripePromise(promiseStripe);
-      }
-    };
-
-    if (props.stripePK) {
-      getStripePromise(props.stripePK);
+    let isMounted = true;
+    if (isMounted) {
+      setClient((client) =>
+        client.project !== project
+          ? createClient({ ...client, project })
+          : client
+      );
     }
-  }, [props.stripePK]);
+    return () => {
+      isMounted = false;
+    };
+  }, [project]);
 
   useEffect(() => {
-    if (stateStripePromise) {
-      setClient(handleCreateClient(stateStripePromise, props.project));
-    }
-  }, [props.project, stateStripePromise]);
+    let isMounted = true;
+    stripePromise?.then((stripe) => {
+      if (isMounted && stripe) {
+        setClient((client) => createClient({ ...client, stripe }));
+      }
+    });
+    return () => {
+      isMounted = false;
+    };
+  }, [stripePromise]);
 
   return (
     <FormspreeContext.Provider value={{ client }}>
-      {props.stripePK ? (
-        <>
-          {stateStripePromise && (
-            <Suspense fallback={<p>....</p>}>
-              <Elements stripe={stateStripePromise}>
-                <>{props.children}</>
-              </Elements>
-            </Suspense>
-          )}
-        </>
+      {stripePromise ? (
+        <Elements stripe={stripePromise}>{children}</Elements>
```

Comment:
`Elements` handle `Promise<Stripe | null> | null | ...`, we can just pass it through and it'll just work.